client.go

```diff
@@ -336,15 +336,22 @@ func (c *Client) ReadDir(p string) ([]os.FileInfo, error) {
 			}
 			count, data := unmarshalUint32(data)
 			for i := uint32(0); i < count; i++ {
-				var filename string
+				var filename, longname string
 				filename, data = unmarshalString(data)
-				_, data = unmarshalString(data) // discard longname
+				// longname contains useful information and should not be discarded
+				// mode         hardlink   user     group      m_time               name
```

Comment:
The second field is not `hardlink` it is `linkcount` there is a subtle difference.